### ISSUE TITLE
Stuck on ngit claim

### ISSUE THREAD

**Issue by Giszmo** [DanConwayDev/ngit-cli#1]
Poking around, trying out things I tried to use ngit with my main repository for WalletScrutiny. It contains 18000 files or so, so maybe that's an issue or "claim" doesn't  do what I suspect it does but my screen looks like this since about 2h:

```
$ ngit claim
✔ name · WalletScrutiny
✔ description (Optional) · The WalletScrutiny.com website and test scripts
login as npub1ty25xevff4eu7wdqxmxl6vzkjfqgg3239p8ntg8y87w9g762k4squ7sly6
✔ password · ********
this may take a few seconds...
searching for your details...
```

**Comment by DanConwayDev** [comment]
Thanks for trying it out. Sorry you didn't get very far.  

I notice that you are using a test npub.

`login as npub1ty...` indicates that it didn't find your profile when you first logged in.

This time it is hanging whilst looking for your metadata.  It should say `logged in as test` or if it is not present on base relays, or cannot connect to base relays with a 20s timeout: `cannot find your account metadata (name, etc) on relays`.

I cannot recreate your issue. Did you try exiting and running the command again?

Did you issue your profile and relay list before or after logging in? before or after running `ngit claim`?

**Comment by DanConwayDev** [comment]
@Giszmo which OS are you using? #3 seems to have the same issue using the windows package. I've just used the linux package.

**Comment by profxadke** [comment]
I am facing the same issue, both `login` and `claim` seemed to hang forever.
```
$ ngit login
login as npub1pf9ux0dmfd3esz6cddkhuah562x04u4hcwvznhed57y65tt3a7eqx9mp64
✔ password · ********
this may take a few seconds...
searching for your details...
```
---
```
$ ngit claim
✔ name · Personal static web pages.
✔ description (Optional) ·  
login as npub1pf9ux0dmfd3esz6cddkhuah562x04u4hcwvznhed57y65tt3a7eqx9mp64
✔ password · ********
this may take a few seconds...
searching for your details...
```
---
Operating System: Linux
Installed via `cargo install ngit`

And, interestingly its consuming most of my CPU power than any other process running on this machine.
![image](https://github.com/DanConwayDev/ngit-cli/assets/151055211/038dc20f-387e-4020-8338-4072bac51849)

**Comment by Giszmo** [comment]
I tried again with the same result. My original login looked like this:

```
$ ngit login
✔ login with nsec (or hex private key) · nsec1*
✔ encrypt with password · *
this may take a few seconds...
cannot find your account metadata (name, etc) on relays
logged in as npub1ty25xevff4eu7wdqxmxl6vzkjfqgg3239p8ntg8y87w9g762k4squ7sly6
```

I might have defined my metadata after that initial login as login now is also stuck on "searching for details".

**Comment by DanConwayDev** [comment]
This should now be at least partially resolved in v0.1.1 which is now live.

You seem to have quite a lot of relays in your relay list which I wasn't anticipating. The tool connects to them simultaneously and couldn't handle it.

It now queues them up and connects to 5 at a time.

I've added some quite verbose messages around this connectivity which I think is too much but better than nothing.

I'm considering using a similar interface to that used to issue events (eg `ngit prs create` or `ngit claim`). What do you think?

I also don't think I need to reach out to every relay just to check your relay list / profile hasn't been updated.

@sectore bumped rust-nostr (dealing with the breaking changes) which I think might have helped.

I haven't implemented the queuing mechanism for sending messages because in my experimentation it didn't seem to cause any issues. Although just trying it again now my fan starts spinning and some relays are only receiving some of the messages.

Thanks for trying out the tool and sorry about this stream of consciousness. 

Your feedback and interest is most welcome.

**Comment by Giszmo** [comment]
Looks better now. It terminates with:

```
maintainers.yaml created. commit and push.
publishing repostory reference...
 ✔  [my-relay] [repo-relay] wss://nos.lol 
 ✔  [my-relay] [repo-relay] wss://nostr-pub.wellorder.net 
 ✘  [my-relay] [repo-relay] wss://nostr.oxtr.dev error: relay not connected (status changed)
 ✔  [my-relay] [repo-relay] wss://nostr.mom 
 ✔  [my-relay] [repo-relay] wss://relay.damus.io 
 ✘  [my-relay] [repo-relay] wss://relay.toastr.space error: blocked: Check out https://toastr.space to get access.
 ✘  [my-relay] [repo-relay] wss://purplepag.es error: blocked: Purple Pages only accepts kind 0, 3 and 10002.
 ✘  [my-relay] [repo-relay] wss://relay.snort.social error: no active subscription
 ✔  [my-relay] [repo-relay] wss://nostr.mutinywallet.com                                                                                                                                                           
```

**Comment by Giszmo** [comment]
And it's listed on https://gitworkshop.dev/ now. Great. Will see what to do with it ...

**Comment by profxadke** [comment]
Adding up a comment, it works on my machine after update.

**Comment by DanConwayDev** [comment]
fixed in 45dc04b81bf7c07f249c9539a48516d4c17abdba, it was the use of futures::executor::block_on that was causing it.
